specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/examples/Patches_Apply.json

```diff
@@ -0,0 +1,101 @@
+    "parameters": {
+      "subcriptionId": "00000000-0000-0000-0000-000000000000",
+      "resourceGroupName": "rg",
+      "builderName": "testBuilder",
+      "patchName": "testPatch-25fe4b",
+      "api-version": "2023-05-02-preview"
+    },
+    "responses": {
+      "200": {
+        "body": {
+          "value": {
+            "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/builders/testBuilder/patches/testPatch-25fe4b",
+            "name": "testPatch-25fe4b",
+            "type": "Microsoft.App/builders/patches",
+            "properties": {
+              "targetEnvironmentId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/managedEnvironments/test-env",
+              "targetContainerAppId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/containerApps/test-app",
+              "targetRevisionId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/apps/test-app/revisions/test-app--jm3vvry",
+              "provisioningState": "Succeeded",
+              "detectionStatus": "Succeeded",
+              "patchApplyStatus": "Succeeded",
+              "patchDetails": [
+                {
+                  "targetContainerName": "test-container",
+                  "targetImageName": "testregistry.azurecr.io/test-image:latest",
+                  "newImageName": "testregistry.azurecr.io/test-image:latest-patched-202210101206",
+                  "newRunImage": {
+                    "name": "mcr.microsoft.com/dotnet/aspnet:7.0.7-cbl-mariner2.0",
+                    "frameworkAndVersion": "dotnet:7.0.7",
+                    "osAndVersion": "cbl-mariner2.0"
+                  },
+                  "oldRunImage": {
+                    "name": "mcr.microsoft.com/dotnet/aspnet:7.0.5-cbl-mariner2.0",
+                    "frameworkAndVersion": "dotnet:7.0.5",
+                    "osAndVersion": "cbl-mariner2.0"
+                  },
+                  "patchType": "FrameworkSecurity"
+                }
+              ]
+            },
+            "systemData": {
+              "createdBy": "sample@microsoft.com",
+              "createdByType": "User",
+              "createdAt": "2022-10-11T11:05:51.4940669Z",
+              "lastModifiedBy": "sample@microsoft.com",
+              "lastModifiedByType": "User",
+              "lastModifiedAt": "2022-10-11T11:05:51.4940669Z"
+            }
+          }
+        }
+      },
+      "202": {
+        "headers": {
+          "Azure-AsyncOperation": "https://management.azure.com/subscriptions/{subscription-id}/providers/Microsoft.App/locations/{location}/operationStatuses/{operationId}"
+        },
+        "body": {
+          "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.App/builders/testBuilder/patches/testPatch/testPatch-25fe4b",
```

Comment:
Missed mentioning this previously, but we do not have to return an object here if we do not want to. The [action method rpc](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests) allows the response to be what we want (adhering to certain guidelines). 

The only restriction seems to be that for async operations we [must return a location header](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously) where the async operation can be monitored.

Reply:
Got it! Thanks for the reminder! I think it would be good to return all these info when the patching operation is running, but we could consider including a revision info when patch has finished and spins up a new revision. Do you see any potential issue with this?

Reply:
I've just went through other APIs samples and specs. It seems like most of the 201 and 200 response bodies are same in terms of the properties returned. Meanwhile, for the portal experience, I think there's no revision info returned when user starts a new revision. I think it could be ok to not return the revision info here.